arroyo/processing/strategies/decoder.py

```diff
@@ -0,0 +1,130 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from functools import partial
+from typing import Callable, Generic, Optional, Sequence, Tuple, TypeVar
+
+import fastjsonschema
+import rapidjson
+
+from arroyo.backends.kafka import KafkaPayload
+from arroyo.processing.strategies.abstract import ProcessingStrategy
+from arroyo.processing.strategies.run_task import RunTaskWithMultiprocessing
+from arroyo.types import Message
+
+T = TypeVar("T")
+
+
+@dataclass(frozen=True)
+class DecodedKafkaMessage(Generic[T]):
+    key: Optional[bytes]
+    decoded: T
+    headers: Sequence[Tuple[str, bytes]]
+
+
+def validation_func(
+    codec: Codec[T], validate: bool, message: Message[KafkaPayload]
+) -> DecodedKafkaMessage[T]:
+    return DecodedKafkaMessage(
+        message.payload.key,
+        codec.decode(message.payload.value, validate),
+        message.payload.headers,
+    )
+
+
+class KafkaMessageDecoder(ProcessingStrategy[KafkaPayload], Generic[T]):
+    """
+    Decode messages to be forwarded to the next step. Optional validation.
+    This strategy accepts a KafkaPayload and only performs validation on the
+    message value. Headers and keys are forwarded without being validated or parsed.
+    """
+
+    def __init__(
+        self,
+        codec: Codec[T],
+        validate: bool,
+        next_step: ProcessingStrategy[DecodedKafkaMessage[T]],
+        num_processes: int,
+        max_batch_size: int,
+        max_batch_time: float,
+        input_block_size: int,
+        output_block_size: int,
+        initializer: Optional[Callable[[], None]] = None,
+    ) -> None:
+        self.__task_runner = RunTaskWithMultiprocessing(
```

Comment:
Why hardcoding multiprocessing here? 
Multiprocessing adds a lot of overhead for cheap tasks on consumers that have a relatively low throughput. Moreover it makes the load on the downstream step very bursty. So often it may be counter productive. What about making it optional ?